plugins/github/server/github.ts

```diff
@@ -1,73 +1,295 @@
-import { createOAuthUserAuth } from "@octokit/auth-oauth-user";
-import find from "lodash/find";
-import { App, Octokit } from "octokit";
-import pluralize from "pluralize";
 import {
-  IntegrationService,
-  IntegrationType,
-  Unfurl,
-  UnfurlResponse,
-} from "@shared/types";
+  createOAuthUserAuth,
+  createAppAuth,
+  type OAuthWebFlowAuthOptions,
+  type InstallationAuthOptions,
+} from "@octokit/auth-app";
+import { Octokit } from "octokit";
+import pluralize from "pluralize";
+import { IntegrationService, IntegrationType, Unfurl } from "@shared/types";
 import Logger from "@server/logging/Logger";
 import { Integration, User } from "@server/models";
 import { GitHubUtils } from "../shared/GitHubUtils";
 import env from "./env";
 
-/**
- * It exposes a GitHub REST client for accessing APIs which
- * particulary require the client to authenticate as a GitHub App
- */
-class GitHubApp {
-  /** Required to authenticate as GitHub App */
-  private static id = env.GITHUB_APP_ID;
-  private static key = env.GITHUB_APP_PRIVATE_KEY
+enum Resource {
+  PR = "pull",
+  Issue = "issue",
+}
+
+type PreviewData = {
+  [Resource.PR]: {
```

Comment:
You can use the `UnfurlResponse` type for these, it's generic so the shape of `meta` can be passed in

Reply:
Resolving types separately altogether in another pass.